webauthn-rs/src/lib.rs

```diff
@@ -577,6 +577,56 @@ impl Webauthn {
             .map(|(ccr, rs)| (ccr, PasskeyRegistration { rs }))
     }
 
+    /// Passkey registration, but with exposed parameters for hints and 
+    /// authenticator attachment, allowing customization of restrictions regarding
+    /// passkey vs platform authenticators.
+    pub fn start_passkey_registration2(
+        &self,
+        user_unique_id: Uuid,
+        user_name: &str,
+        user_display_name: &str,
+        exclude_credentials: Option<Vec<CredentialID>>,
+        hints: Option<Vec<PublicKeyCredentialHints>>,
```

Comment:
`hints` doesn't need to be exposed as a parameter. By default, passkey should add hint as `hybrid` and same goes for security key where the hint should be `security-key` and authenticatorAttachment for both should be `cross-platform` 

you can checkout https://opotonniee.github.io/webauthn-playground/ for reference.

Reply:
This is used for both passkey and platform authenticators in the SL-side PR, which require different hints and authenticator attachments.

Reply:
Ah, that makes sense.

Reply:
no?

Its used for both UnspecifiedWebauthn (ui_hint_authenticator_attachment is None), and SecurityKey (ui_hint_authenticator_attachment is Some(cross-platform) I think)

Reply:
yep - my bad. makes sense.